onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.h

```diff
@@ -25,9 +25,8 @@ Status AddScalarOperand(ModelBuilder& model_builder, std::vector<uint32_t>& inpu
 // adds ANEURALNETWORKS_TRANSPOSE operation
 Status AddNnapiTranspose(ModelBuilder& model_builder,
                          const std::string& data_input,
-                         const std::string& perm_input,
-                         const std::vector<int32_t>& perm,
-                         const std::string& output);
+                         const std::string& perm_input, const gsl::span<const int32_t> perm,
+                         const std::string& output, const Shape* output_shape);
 
```

Comment:
Is this called from anywhere other than AddToModelBuilderImpl? If not the output_shape is never nullptr and doesn't need to be passed as a pointer here.

Reply:
I think this optional output_shape is put here mainly for reusing it for both `output shape provided` and `output shape needs to computed` cases... e.g. for AddNnapiReshape called in BuildBatchMatMul:

https://github.com/microsoft/onnxruntime/blob/ebff15d74394b8549a5af0f5f772073a928b5a3a/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.cc#L209

AddNnapiTranspose is just trying to do something similar here.